components/ai_chat/core/browser/model_validator.h

```diff
@@ -0,0 +1,43 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_AI_CHAT_CORE_BROWSER_MODEL_VALIDATOR_H_
+#define BRAVE_COMPONENTS_AI_CHAT_CORE_BROWSER_MODEL_VALIDATOR_H_
+
+#include <optional>
+
+#include "brave/components/ai_chat/core/browser/ai_chat_credential_manager.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom-forward.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom-shared.h"
+#include "brave/components/ai_chat/core/common/mojom/settings_helper.mojom.h"
+#include "url/url_constants.h"
+
+namespace ai_chat {
+
+constexpr size_t kMinCustomModelContextSize = 1;
+constexpr size_t kMaxCustomModelContextSize = 2'000'000;
+constexpr size_t kDefaultCustomModelContextSize = 4000;
```

Comment:
An initial upper-boundary of 2,000,000 tokens was chosen for now. This is somewhat arbitrary, but based on some of the highest known context sizes in the ecosystem today. We will almost certainly have to revisit this at a later point.